sql/analyzer/pushdown.go

```diff
@@ -460,6 +460,9 @@ func getPredicateExprsHandledByLookup(ctx *sql.Context, a *Analyzer, idxTable *p
 	if !ok {
 		return nil, nil
 	}
+	if filteredIdx.IsSpatial() {
```

Comment:
are spatial lookups lossy? what scenarios do we need to keep filter vs not?

Reply:
Spatial lookups are lossy, so we definitely should keep the filter.
They're basically doing a cheap bounding box check, before running in actual expensive st_intersect/st_equals/etc function.

Reply:
it would be good to make this info and its life cycle peculiarities more obvious

Reply:
does this mean add a comment?